packages/sdk/src/interaction/interaction.js

```diff
@@ -149,9 +159,19 @@ export const prepAccount = (acct, opts = {}) => ix => {
   const role = opts.role
   const tempId = uuid()
 
-  if (acct.authorization && isFn(acct.authorization)) acct = {resolve: acct.authorization}
+  if (acct.authorization && isFn(acct.authorization))
+    acct = {resolve: acct.authorization}
   if (!acct.authorization && isFn(acct)) acct = {resolve: acct}
 
+  const resolve = acct.resolve
+  if (resolve)
```

Comment:
Yup, if acct is provided as a data structure as oppose to a resolver/authorization function then it will be falsey.  To handle that instance, `acct` is piped through `prepAccountKeyId(acct)` on the line below
https://github.com/onflow/fcl-js/blob/cd2141b78b2a6bb34a14896e4298114e369780aa/packages/sdk/src/interaction/interaction.js#L173
This insures that the account key is prepped if the object variant of account was provided instead of the resolver function.

Reply:
It might be if authz is passed as a data structure and not a function that resolves to a data structure as per the [spec](https://github.com/onflow/fcl-js/blob/cd2141b78b2a6bb34a14896e4298114e369780aa/packages/fcl/src/wallet-provider-spec/authorization-function.md).  That case is handled on the next line https://github.com/onflow/fcl-js/blob/cd2141b78b2a6bb34a14896e4298114e369780aa/packages/sdk/src/interaction/interaction.js#L173